sdk/src/program-manager.ts

```diff
@@ -898,41 +1027,53 @@ class ProgramManager {
+     * @param {string} staker_address Address of the staker who is unbonding the credits
      * @param {number} amount Amount of credits to unbond. If the address of the executor of this function is an
      * existing validator, it will subtract this amount of credits from the validator's staked credits. If there are
      * less than 1,000,000 credits staked pool after the unbond, the validator will be removed from the validator set.
      * If the address of the executor of this function is not a validator and has credits bonded as a delegator, it will
      * subtract this amount of credits from the delegator's staked credits. If there are less than 10 credits bonded
      * after the unbond operation, the delegator will be removed from the validator's staking pool.
-     * @param {Options} options Options for the execution
+     * @param {ExecuteOptions} options Options for the execution
      */
-    async unbondPublic(amount: number, options: Partial<ExecuteOptions> = {}): Promise<string | Error> {
-        const tx = <Transaction>await this.buildUnbondPublicTransaction(amount, options);
+    async unbondPublic(staker_address: string, amount: number, options: Partial<ExecuteOptions> = {}): Promise<string | Error> {
+        const tx = <Transaction>await this.buildUnbondPublicTransaction(staker_address, amount, options);
         return await this.networkClient.submitTransaction(tx);
     }
 
     /**
      * Build a transaction to claim unbonded public credits in the Aleo network.
      *
+     * @param {string} staker_address - The address of the staker who is claiming the credits.
      * @param {Partial<ExecuteOptions>} options - Override default execution options.
      * @returns {Promise<Transaction | Error>} - A promise that resolves to the transaction or an error message.
      *
      * @example
+     * // Create a keyProvider to handle key management
+     * const keyProvider = new AleoKeyProvider();
+     * keyProvider.useCache = true;
+     *
+     * // Create a new ProgramManager with the key that will be used to claim unbonded credits.
      * const programManager = new ProgramManager("https://api.explorer.aleo.org/v1", keyProvider, undefined);
-     * const transaction = await programManager.buildClaimUnbondPublicTransaction();
-     * console.log(transaction);
+     *
+     * // Create the claim unbonded transaction object for later use.
+     * const tx = await programManager.buildClaimUnbondPublicTransaction("aleo1jx8s4dvjepculny4wfrzwyhs3tlyv65r58ns3g6q2gm2esh7ps8sqy9s5j");
+     * console.log(tx);
+     *
+     * // The transaction can be submitted later to the network using the network client.
+     * programManager.networkClient.submitTransaction(tx);
      */
-    async buildClaimUnbondPublicTransaction(options: Partial<ExecuteOptions> = {}): Promise<Transaction | Error> {
+    async buildClaimUnbondPublicTransaction(staker_address: string, options: Partial<ExecuteOptions> = {}): Promise<Transaction | Error> {
```

Comment:
The emphasize which address is being used. In general most of the methods that mirror credits.aleo have the parameters specified outside of the ExecutionOptions so that it fits the same form factor as the Aleo Instructions/Leo methods. For all credits.aleo transactions it's extremely important to get inputs right, so they're structured this way to force the caller to consider the options properly/fit the form factor of Aleo instructions.

Reply:
In general I think we should move toward an API where everything is specified with options objects. That makes it *much* easier to add new fields over time, and also makes the API clearer for users.

But that can be done in a future PR, so I'll accept this for now.